_search-plugins/knn/performance-tuning.md

```diff
@@ -32,31 +32,86 @@ Take the following steps to improve indexing performance, especially when you pl
-
-* **Increase the number of indexing threads**
-
-   If the hardware you choose has multiple cores, you can allow multiple threads in native library index construction by speeding up the indexing process. Determine the number of threads to allot with the [knn.algo_param.index_thread_qty]({{site.url}}{{site.baseurl}}/search-plugins/knn/settings#cluster-settings) setting.
-
-  Keep an eye on CPU utilization and choose the correct number of threads. Because native library index construction is costly, having multiple threads can cause additional CPU load.
+Make sure to reenable `refresh_interval` after indexing finishes.
+
+### Disable replicas (no OpenSearch replica shard)
+
+Set replicas to `0` to prevent duplicate construction of native library indexes in both primary and replica shards. When you enable replicas after indexing finishes, the serialized native library indexes are directly copied. If you have no replicas, losing nodes might cause data loss, so it's important that the data lives elsewhere so this initial load can be retried in case of an issue.
+
+### Increase the number of indexing threads
+
+If the hardware you choose has multiple cores, you can allow multiple threads in native library index construction by speeding up the indexing process. Determine the number of threads to allot with the [knn.algo_param.index_thread_qty]({{site.url}}{{site.baseurl}}/search-plugins/knn/settings#cluster-settings) setting.
+
+Keep an eye on CPU utilization and choose the correct number of threads. Because native library index construction is costly, having multiple threads can cause additional CPU load.
+
+
+## (Expert-level) Disable storing of vector fields in source
+
+In OpenSearch, the JSON document body that is passed during indexing stored is stored in `_source` field of the document. The field is not indexed or searchable, but returned in the response inside the `_source` key. To improve performance, uou can remove the vector field from this `_source` to save up the disk space, as shown in the following index mapping where the `location` vector field is not added to the source:
+
+  ```json
+  PUT /<index_name>/_mappings
+  {
+      "_source": {
+        "excludes": ["location"]
+      },
+      "properties": {
+          "location": {
+              "type": "knn_vector",
+              "dimension": 2,
+              "method": {
+                  "name": "hnsw",
+                  "space_type": "l2",
+                  "engine": "faiss"
+              }
+          }
+      }
+  }
+  ```
+
+  
+After disabling the `_source` for the vector field API like update, update_by_query, reindex will not as expected because OpenSearch has no way to know what was the vector.
```

Comment:
How is `update_by_query` related to the removal of the vector field and the reindex operation? 

```suggestion
After disabling the `_source` for the vector field API like update, update_by_query, reindex will not as expected because OpenSearch has no way to know what was the vector.
```

Reply:
The side effects is coming from removal of a field from _source. In this case because that field is vector field.

Reply:
So `update` and `reindex` operations involving the disabled field will not work?

Reply:
Trying to understand what the side effects you are referring to are.

Reply:
pushed a commit.

Reply:
That still doesn't clarify the relation between the "update_by_query" operation, the missing vector, and reindexing. 

Would the following rewording be accurate? 

"After disabling the `_source` vector field, vector API operations, such as `update_by_query` or `reindex`, will not operate since OpenSearch can no longer find the vector".

Reply:
The behavior is true for any field and not related to vector field. We say the same thing here in this PR too: [_field-types/metadata-fields/source.md](https://github.com/opensearch-project/documentation-website/pull/6933/files#diff-9a227ae77d563b0eb8cb9cdc1f5bacb86f7bb000d6768162c34848aa11aa0a97)

I am just referring the same thing here because the behavior is same.